pkg/pillar/cmd/zedagent/parseconfig.go

```diff
@@ -3199,3 +3207,53 @@ func handleDeviceOperation(ctxPtr *zedagentContext, op types.DeviceOperation) {
 	shutdownAppsGlobal(ctxPtr)
 	// nothing else to be done
 }
+
+func parseEdgeNodeClusterConfig(getconfigCtx *getconfigContext,
+	config *zconfig.EdgeDevConfig) {
+
+	ctx := getconfigCtx.zedagentCtx
+	zcfgCluster := config.GetCluster()
+	if zcfgCluster == nil {
+		log.Functionf("parseEdgeNodeClusterConfig: No EdgeNodeClusterConfig, Unpublishing")
+		pub := ctx.pubEdgeNodeClusterConfig
+		items := pub.GetAll()
+		if len(items) > 0 {
+			log.Functionf("parseEdgeNodeClusterConfig: Unpublishing EdgeNodeClusterConfig")
+			ctx.pubEdgeNodeClusterConfig.Unpublish("global")
+		}
+		return
+	}
+	ipAddr, ipNet, err := net.ParseCIDR(zcfgCluster.GetClusterIpPrefix())
+	if err != nil {
+		log.Errorf("parseEdgeNodeClusterConfig: ParseCIDR failed %s", err)
+		return
+	}
+	ipNet.IP = ipAddr
+
+	joinServerIP := net.ParseIP(zcfgCluster.GetJoinServerIp())
```

Comment:
Missing error check?

Reply:
Ok, will add nil check for the 'joinServerIP'.